### ISSUE TITLE
haxelib selfupdate fails

### ISSUE THREAD

**Issue by ibilon** [HaxeFoundation/haxelib#189]
I tried to run `haxelib selfupdate` but ran into this issue:
```
Error writing file /usr/lib/haxe/haxelib. Please ensure you have write permissions. 
  [file_open,/usr/lib/haxe/haxelib]
```
I ran the command as root (linux 64) and I have the permission.

Doing a copy of `/usr/lib/haxe/haxelib` and using it to run the command fixed it.

**Comment by ibilon** [comment]
Oh and btw trying a second time `haxelib selfupdate` will say `Haxelib was already up to date...` and assume everything went right the last time and won't try to update.

**Comment by madneon** [comment]
I can confirm this error on Ubuntu 14.04 (64).

```bash
$ haxelib selfupdate
Haxelib was already up to date...
Error writing file /usr/lib/haxe/haxelib. Please ensure you have write permissions. 
  [file_open,/usr/lib/haxe/haxelib]
```

**Comment by andyli** [comment]
What is the haxelib version you were using? The selfupdate action is performed by the old version of haxelib.

**Comment by ibilon** [comment]
@andyli 3.2.0-rc.1, the one bundled with haxe 3.2.0-rc.2

**Comment by facchione** [comment]
Following @ibilon , on ubuntu 12.04, I was trying to update 
    `Haxe Library Manager 3.1.0-rc.4 - (c)2006-2013 Haxe Foundation`
using
   `$ haxelib selfupdate`
and getting 
`Error writing file /usr/lib/haxe/haxelib. Please ensure you have write permissions. `

I guess it's having trouble over-writing itself, a running executable, which is why even running as root fails?  Copying my executable to a new name and running that, which then updates the real `haxelib` seems to work:

    `$ cp /usr/lib/haxe/haxelib /usr/lib/haxe/haxelib.bak`
    `$ /usr/lib/haxe/haxelib.bak selfupdate`

**Comment by nadako** [comment]
Fixed by #291, now `selfupdate` doesn't mess with the global executable.
